### PR TITLE
changes drop calculation to not use seeded stats #7929

### PR DESCRIPTION
**Pull request by gambtho**
Fixes https://github.com/HabitRPG/habitrpg/issues/7929
### Changes

Uses math.random instead of predictable random from drop calculations

---

UUID: 349ea7be-f378-4593-8702-672f75fa83f9

### CODE REVIEW

File: common/script/fns/randomDrop.js [hunk 1]
```diff
@@ -21,7 +20,7 @@ module.exports = function randomDrop (user, options, req = {}) {
   let rarity;
   let task;
 
-  let predictableRandom = options.predictableRandom || _predictableRandom;
+  let random = options.predictableRandom || Math.random;
```
Comment (paglias) [inline common/script/fns/randomDrop.js]: is this option used? if yes, what's being passed to it?

Comment (gambtho) [reply common/script/fns/randomDrop.js]: i believe it's only used for testing.

**Comment by paglias** [comment]
randomVal is also used in `revive`, we should check that it's only used on the server and never on the client. Also we should be able to remove `predictableRandom` at all from the code

**Comment by paglias** [comment]
or better, never used anywhere where we want to run the same op and expect the same result between server and client

File: common/script/fns/randomDrop.js (hunk 1 above)
Comment (paglias) [reply common/script/fns/randomDrop.js]: 👍

**Comment by gambtho** [comment]
removing predictable random completely will require changes to a significant amount of tests and functions, not sure it's worth the trouble

**Comment by paglias** [comment]
well, if it's not used I'd like to remove it, we don't want to keep unused code that months from now we won't remember why it's there

**Comment by gambtho** [comment]
Totally fair point -- just a bigger change.    Adjusting removing predictableRandom from randomVal   has an impact to a number of the tests that rely on it.

**Comment by paglias** [comment]
Yeah I understand... Maybe we could tweak the predictable random function
directly to allow the use of Math.random as an option?

Il 03 set 2016 20:04, "Tom Gamble" notifications@github.com ha scritto:

**Comment by gambtho** [comment]
Ooh, that's a cool idea.   Easy way to maintain the existing structure, but add randomness.   I'll revert my original change and try that.

**Comment by gambtho** [comment]
@paglias thanks for the feedback, updated PR makes drops random, but minimizes other side effects

**Comment by paglias** [comment]
I think you can keep the old tests, they're still valid for the predictable behavior.

@crookedneighbor opinions?

**Comment by crookedneighbor** [comment]
I don't like this. Defaulting to `Math.random` in a function named `predicatableRandom` doesn't make sense to me. 

I think instead we should do some dependency injection into the randomVal function. One of the options should be a function to produce a random value, and if none is passed in, it should default to `Math.random`.

**Comment by gambtho** [comment]
Totally fair point -- i'll update and resubmit

**Comment by gambtho** [comment]
tried a few different ways to do this, please let me know what you think

**Comment by crookedneighbor** [comment]
Did you intend to close this?

**Comment by gambtho** [comment]
got concerned with the number of commits above -- wanted to validate I hadn't pulled the wrong thing

**Pull request closed without merging**